src/main/java/com/google/sps/data/OrganizationUpdater.java

```diff
@@ -0,0 +1,108 @@
+
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.Set;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Arrays;
+import com.google.appengine.api.datastore.Entity;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+
+public final class OrganizationUpdater {
+
+  private Entity entity;
+
+  public OrganizationUpdater(Entity entity) {
+     this.entity = entity;
+  }
+
+  public Entity getEntity() {
+      return this.entity;
+  }
+
+  public void updateOrganization(HttpServletRequest request, boolean isMaintainer, boolean isModerator) throws IllegalArgumentException{
+    Set<String> requiresMaintainer = new HashSet<String>();
+    Set<String> requiresModerator = new HashSet<String>();
+    Map<String, String> properties = new HashMap<String,String>();
+
+    requiresMaintainer.add("isApproved");
+    requiresModerator.add("moderatorList");
+
+    // Format is Form Name, Entity Property 
+    properties.put("org-name", "orgName");
+    properties.put("org-email", "orgEmail");
+    properties.put("org-street-address", "orgStreetAddress");
+    properties.put("org-phone-num", "orgPhoneNum");
+    properties.put("org-url", "orgUrl");
+    properties.put("org-description","orgDescription");
+    properties.put("approval", "isApproved");
+    properties.put("moderator-list", "moderatorList");
+
+    for(Map.Entry<String, String> entry : properties.entrySet()) {
+      String formKey = entry.getKey();
+      String propertyKey = entry.getValue();
+      String formValue = "";
+
+      // will only get approval param is a maintainer sent the request
+      if(requiresMaintainer.contains(propertyKey) && !isMaintainer) {
+        continue;
+      }
+
+      // will only get moderatorList param if request is sent by a moderator or maintainer
+      if(requiresModerator.contains(propertyKey) && !isModerator && !isMaintainer) {
+        continue;
+      }
+
+      // setting organization description is optional
+      if(!propertyKey.equals("orgDescription")) {
+        try {
+          formValue = getParameterOrThrow(request, formKey);
+        } catch(IllegalArgumentException err) {
+          throw new IllegalArgumentException();
+        }
+      } else {
+        formValue = request.getParameter(formKey);
+      }
+      setOrganizationProperty(propertyKey, formValue);
+    }
+  }
+
+  private String getParameterOrThrow(HttpServletRequest request, String formKey) {
+    String result = request.getParameter(formKey);
+    if(result.isEmpty() || result == null) {
+        throw new IllegalArgumentException("Form value cannot be null");
+    }
+    return result;
+  }
+
+  private void setOrganizationProperty(String propertyKey, String formValue) {
+    if(propertyKey.equals("moderatorList")) {
+      ArrayList<String> newModeratorList = new ArrayList<String>(Arrays.asList(formValue.split("\\s*,\\s*")));
+      this.entity.setProperty("moderatorList", newModeratorList);
```

Comment:
Is this right? Aren't you storing userids in the db, and accepting emails in the form?

Reply:
Good catch here.

Reply:
Now will be using givr user translate email to ids function